Src/Newtonsoft.Json/Utilities/BoxedPrimitives.cs

```diff
@@ -87,9 +90,42 @@ internal static class BoxedPrimitives
         internal static readonly object Int64_7 = 7L;
         internal static readonly object Int64_8 = 8L;
 
-        internal static object Get(decimal value) => value == decimal.Zero ? DecimalZero : value;
+        internal static object Get(decimal value)
+        {
+            // Decimals can contain trailing zeros. For example 1 vs 1.0. Unfortunatly, Equals doesn't check for trailing zeros.
+            // There isn't a way to find out if a decimal has trailing zeros in older frameworks without calling ToString.
```

Comment:
Check this  https://github.com/graphql-dotnet/graphql-dotnet/search?q=DecimalData&type=
Should work for all platforms.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.as?view=net-7.0

This would only be available for the net6.0 target, which already has `decimal.GetBits`. Anyway, I prefer to use public API for getting these values.

Reply:
?
https://www.nuget.org/packages/System.Runtime.CompilerServices.Unsafe/7.0.0-preview.2.22152.2#supportedframeworks-body-tab

netstandard2.0 compatible

Reply:
I won't introduce a dependency for a small perf optimization